### PR TITLE
Add ICU version check to ensure minimum of version 61 is used.

### PR DESCRIPTION
**Pull request by spevans**

### CODE REVIEW

**Comment by spevans** [comment]
Note this should currently fail CI testing due to SR-9699

**Comment by spevans** [comment]
@swift-ci test

**Comment by spevans** [comment]
cc @compnerd What ICU version do you have with Windows? Or are you compiling it from the ICU source tree as part of the build?

**Comment by spevans** [comment]
@compnerd FYI, there are some tests that are broken with ICU63, mostly to do with `currencyCode`, so if you see those tests failing its not a windows issue but something we need to fix in the tests anyway.

**Comment by compnerd** [comment]
Thanks for the heads up, I'm still trying to get Foundation to build, let alone test though.  Pretty close, `NSData`, `NSFileManager` (which is why I am anxious to get the `NSData` work in, so all the focus goes to the `NSFileManager`).

**Comment by spevans** [comment]
@swift-ci test

**Comment by spevans** [comment]
@compnerd @millenomi Any objections to merging this to prevent future ICU nonsense?

File: CoreFoundation/CMakeLists.txt [hunk 1]
```diff
@@ -378,6 +378,12 @@ if(NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
                              PRIVATE
                                ${CURL_INCLUDE_DIRS})
   find_package(ICU COMPONENTS uc i18n REQUIRED)
+
+  set(ICU_MINIMUM_VERSION 61)
+  if(ICU_VERSION VERSION_LESS ${ICU_MINIMUM_VERSION})
+    message(FATAL_ERROR "ICU version ${ICU_VERSION} is less than required version ${ICU_MINIMUM_VERSION}")
+  endif()
+  message(STATUS, "ICU_INCLUDE_DIR: ${ICU_INCLUDE_DIR}")
```
Comment (compnerd) [inline CoreFoundation/CMakeLists.txt]: Please remove the `,`.

**Comment by spevans** [comment]
@swift-ci test

**Comment by spevans** [comment]
@swift-ci test and merge

**Comment by spevans** [comment]
@swift-ci test and merge

**Pull request merged**